### PR TITLE
1.13.1

### PR DESCRIPTION
**Pull request by modmuss50**
Please review current state

Find MCP yourself I dont agree with re-hosting it, this should be changed
Compiles, loads into a world. No mods have been tested

Still to do:

- [ ] Fix access transformers
- [ ] Move to srg, pls
- [ ] Fix test mod
- [ ] Update mappings

### CODE REVIEW

**Comment by modmuss50** [comment]
@Chocohead yeah it is lol 👍 nice spot

File: src/main/java/org/dimdev/rift/util/ArmorMaterialImpl.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public ArmorMaterialImpl(String name, int maxDamageFactor, int[] damageReduction
 
     @Override
     public int getDurability(EntityEquipmentSlot slot) {
-        return ArmorMaterial.MAX_DAMAGE_ARRAY[slot.getIndex()] * maxDamageFactor;
+        return getDurability(slot) * maxDamageFactor;
```
Comment (asiekierka) [inline src/main/java/org/dimdev/rift/util/ArmorMaterialImpl.java]: Won't this just call itself recursively?

File: src/main/java/org/dimdev/rift/network/Message.java [hunk 2]
```diff
@@ -13,13 +13,14 @@
...
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.registry.IRegistry;
 import net.minecraft.util.registry.RegistryNamespaced;
 import net.minecraft.world.WorldServer;
 
 import java.util.function.Predicate;
 
 public abstract class Message {
-    public static final RegistryNamespaced<ResourceLocation, Class<? extends Message>> REGISTRY = new RegistryNamespaced<>();
```
Comment (asiekierka) [inline src/main/java/org/dimdev/rift/network/Message.java:22]: Should the "message" string not be namespaced?

Comment (modmuss50) [reply src/main/java/org/dimdev/rift/network/Message.java:22]: Should be better, I wasn't sure if it was best to add a new class or use mixin's to add the method straight to IRegistry

**Comment by ghost** [comment]
How does setting up MCP for 1.13.1 work

**Comment by Chocohead** [comment]
I've got a half finished version of FG2 that is designed to take the newer FG3 config style [here](https://github.com/Chocohead/ForgeGradle), theoretically all that needs finishing is `GenSrgs#writeOutExcs` be switched to pull from the `constructors.txt` rather than the old `joined.exc`. That would avoid any re-hosting of MCP which rubs some people the wrong way and work as a stop gap until FG3 actually works and MixinGradle plays ball with it.
I've not tested it actually trying to create a clean workspace, but it is much closer to work with the new style than stock FG2 is. Might even be able to allow Rift to have `setupDecompWorkspace` back if we're lucky.

**Comment by modmuss50** [comment]
@Chocohead is taking this PR over, see #58

**Pull request closed without merging**